packages/squiggle-lang/src/value/valueList.ts

```diff
@@ -0,0 +1,267 @@
+      uniqueArray.push(item);
+    }
+  }
+
+  return uniqueArray;
+}
+
+export function uniqBy(array: Value[], lambda: Lambda, context: any): Value[] {
+  _expect1ParamsLength(lambda, 1);
+  const seen: Value[] = [];
+  const result: Value[] = [];
+
+  for (const item of array) {
+    const computed = lambda.call([item], context);
+    if (!isUniqableType(computed)) {
+      throw new REOther(
+        `Can't apply uniq() to element with type ${computed.type}`
+      );
+    }
+    if (!seen.some((existingItem) => isEqual(existingItem, computed))) {
+      seen.push(computed);
+      result.push(item);
+    }
+  }
+
+  return result;
+}
+
+export function filter(array: Value[], lambda: Lambda, context: any): Value[] {
+  _expect1ParamsLength(lambda, 1);
+  return array.filter(_binaryLambdaCheck1(lambda, context));
+}
+
+export function every(array: Value[], lambda: Lambda, context: any): boolean {
+  _expect1ParamsLength(lambda, 1);
+  return array.every(_binaryLambdaCheck1(lambda, context));
+}
+
+export function some(array: Value[], lambda: Lambda, context: any): boolean {
+  _expect1ParamsLength(lambda, 1);
+  return array.some(_binaryLambdaCheck1(lambda, context));
+}
+
+export function find(array: Value[], lambda: Lambda, context: any): Value {
+  _expect1ParamsLength(lambda, 1);
+  return _findArrayElement(array, lambda, context);
+}
+
+export function findIndex(
+  array: Value[],
+  lambda: Lambda,
+  context: any
+): number {
+  return _findArrayIndex(array, lambda, context);
+}
+
+export function map(array: Value[], lambda: Lambda, context: any): Value[] {
+  const mapped: Value[] = new Array(array.length);
+  const parameters = lambda.getParameterNames().length;
+
+  // this code is intentionally duplicated for performance reasons
+  if (parameters === 1) {
+    for (let i = 0; i < array.length; i++) {
+      mapped[i] = lambda.call([array[i]], context);
+    }
+  } else if (parameters === 2) {
+    for (let i = 0; i < array.length; i++) {
+      mapped[i] = lambda.call([array[i], vNumber(i)], context);
+    }
+  } else {
+    throw new REOther("Expected function with 1 or 2 parameters");
+  }
+  return mapped;
+}
+
+export function reduce(
+  array: Value[],
+  initialValue: Value,
+  lambda: Lambda,
+  context: any
+): Value {
+  const parameters = lambda.getParameterNames().length;
+  if (parameters === 2) {
+    return array.reduce(
+      (acc, elem) => lambda.call([acc, elem], context),
+      initialValue
+    );
+  } else if (parameters === 3) {
+    return array.reduce(
+      (acc, elem, index) => lambda.call([acc, elem, vNumber(index)], context),
+      initialValue
+    );
+  } else {
+    throw new REOther("Expected function with 2 or 3 parameters");
+  }
+}
+
+export function reduceReverse(
+  array: Value[],
+  initialValue: Value,
+  lambda: Lambda,
+  context: any
+): Value {
+  return reduce([...array].reverse(), initialValue, lambda, context);
```

Comment:
Note that now, reduceReverse() can take in lambdas with two params, which it couldn't before.